packages/vats/src/vat-chainStorage.js

```diff
@@ -1,30 +1,57 @@
 // @ts-check
-import { Far } from '@endo/far';
+import { Fail } from '@agoric/assert';
+import { E, Far } from '@endo/far';
+import { makeBridgeManager } from './bridge.js';
 import { makeChainStorageRoot } from './lib-chainStorage.js';
 
 export function buildRootObject(vatPowers) {
+  const { D } = vatPowers;
+  D || Fail`D missing in vatPowers ${Object.keys(vatPowers)}`;
+
+  /** @type {Map<BridgeDevice, import('./types.js').BridgeManager>} */
+  const bridgeToManager = new Map();
+
+  /**
+   * Obtain the single manager associated with a bridge device.
+   *
+   * @param {BridgeDevice} bridge
+   * @returns {import('./types.js').BridgeManager}
+   */
+  const provideManagerForBridge = bridge => {
+    let bridgeManager = bridgeToManager.get(bridge);
+    if (!bridgeManager) {
+      bridgeManager = makeBridgeManager(E, D, bridge);
+      bridgeToManager.set(bridge, bridgeManager);
+    }
+    return bridgeManager;
+  };
+
   /**
    * @param {BridgeDevice} bridge
    * @param {string} bridgeId
    * @param {string} rootPath must be unique (caller responsibility to ensure)
    * @param {object} [options]
    */
-  function makeBridgedChainStorageRoot(bridge, bridgeId, rootPath, options) {
-    const { D } = vatPowers;
-    assert(D, `D missing in vatPowers ${Object.keys(vatPowers)}`);
+  const makeBridgedChainStorageRoot = (bridge, bridgeId, rootPath, options) => {
     // Note that the uniqueness of rootPath is not validated here,
     // and is instead the responsibility of callers.
-    const toStorage = message => D(bridge).callOutbound(bridgeId, message);
+
+    const bridgeManager = provideManagerForBridge(bridge);
+    const toStorage = obj => E(bridgeManager).toBridge(bridgeId, obj);
```

Comment:
Why change from `message` to `obj`?

Reply:
Probably habit.  I tend to use `obj` for jsonable values, and `message` for strings.  But that's just me.  I'll put it back.